pkg/api/api.go

```diff
@@ -239,7 +239,7 @@ func (a *API) RegisterRuntimeConfig(runtimeConfigHandler http.HandlerFunc) {
 func (a *API) RegisterDistributor(d *distributor.Distributor, pushConfig distributor.Config) {
 	distributorpb.RegisterDistributorServer(a.server.GRPC, d)
 
-	wrappedDistributor := a.cfg.wrapDistributorPush(d)
+	wrappedDistributor := d.PrePushForwardingMiddleware(a.cfg.wrapDistributorPush(d))
```

Comment:
I think that we need to update the docs on the `cfg.DistributorPushWrapper` now stating that this wrapper doesn't get all the pushed series, but only the ones that are going to be ingested throught he ingesters.

(I also would re-think twice whether that's what we want :thinking:)

Reply:
> I think that we need to update the docs on the cfg.DistributorPushWrapper now stating that this wrapper doesn't get all the pushed series, but only the ones that are going to be ingested throught he ingesters.

Thx, I'll update the docs

> (I also would re-think twice whether that's what we want 🤔)

You're right, afaict this is used by business intelligence, I'll reach out to them after merging this about how they want this to be handled. Keep in mind that the aggregation results will go through another distributor, at which point the aggregation results would then be accounting for by the business intelligence wrapper.

Reply:
fixed https://github.com/grafana/mimir/pull/2295/commits/e551b42e82a069c96580c1993c867b05ed9601fb